Fix/metadata UI improvements

novakzaballa opened the pull request:
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

- Add errors and validations in features and environments settings.
- Improve multiline type metadata.
- Change URL validation type.
- Improve hide metadata.

## How did you test this code?

<!-- If the answer is manually, please include a quick step-by-step on how to test this PR. -->

_Please describe._

kyle-ssg commented on frontend/common/stores/project-store.js:
```diff
@@ -84,15 +84,22 @@ const controller = {
+      })
+      .catch((e) => {
+        e.json().then((result) => {
+          toast(result.metadata[0], 'danger')
```
This assumes result.metadata isn't null.

novakzaballa replied:
Corrected

kyle-ssg commented on frontend/web/components/metadata/AddMetadataToEntity.tsx:
```diff
@@ -334,7 +335,35 @@ const MetadataRow: FC<MetadataRowType> = ({
+      ) : metadata?.type === 'multiline_str' ? (
+        <Flex className='flex-row' style={{ minWidth: '300px' }}>
+          <InputGroup
+            textarea
```
cant you just adjust this single prop rather than everything?

novakzaballa replied:
Updated

kyle-ssg commented on frontend/web/components/modals/CreateFlag.js:
```diff
@@ -1961,12 +1961,20 @@ const FeatureProvider = (WrappedComponent) => {
-            }`,
-          )
+        ({ changeRequest, createdFlag, error, isCreate } = {}) => {
+          if (error?.data?.metadata) {
```
error could be defined but not error?.data?.metadata. With this code it'll show a success message

novakzaballa replied:
Updated

kyle-ssg commented on frontend/common/utils/utils.tsx:
```diff
@@ -382,6 +382,10 @@ const Utils = Object.assign({}, require('./base/_utils'), {
         valid = isEnterprise
         break
       }
+      case 'METADATA': {
```
Note to self: this affects https://github.com/Flagsmith/flagsmith/pull/4281

kyle-ssg commented on frontend/common/stores/project-store.js:
```diff
@@ -84,15 +84,30 @@ const controller = {
+      .catch((e) => {
+        e.json()
+          .then((result) => {
+            if (result.metadata && result.metadata[0]) {
```
1 thing that makes this a


```
(result: Environment) => { // Types Environment
   if(result?.metadata?.[0])
}
```

novakzaballa replied:
done

novakzaballa commented on frontend/common/utils/utils.tsx:
```diff
@@ -396,6 +400,8 @@ const Utils = Object.assign({}, require('./base/_utils'), {
         : null
       : AccountStore.getPlans()
 
+    console.log('DEBUG: plans:', plans)
```
Done

kyle-ssg commented:
As discussed, raised an overall issue [here](https://github.com/Flagsmith/flagsmith/issues/4408). For now there's a few things to tidy up + default custom fields to expanded

novakzaballa commented:
> As discussed, raised an overall issue [here](https://github.com/Flagsmith/flagsmith/issues/4408). For now there's a few things to tidy up + default custom fields to expanded

@kyle-ssg, I adjusted the Vertical alignment, removed the placeholder, and custom fields expanded by default.
I added a screenshot with the UI changes bellow.

![Screenshot 2024-07-29 at 11 34 43 AM](https://github.com/user-attachments/assets/b8b6366c-6b96-4141-9233-82bca184ae97)

kyle-ssg commented:
So much better
